runtime/client/queue.go

```diff
@@ -111,6 +112,12 @@ func (m *MockQueue) GetArtifact_SignedURL(taskID, runID, name string, duration t
 	return args.Get(0).(*url.URL), args.Error(1)
 }
 
+// Task is a mock implementation of github.com/taskcluster/taskcluster-client-go/tcqueue.Task
+func (m *MockQueue) Task(taskID string) (*tcqueue.TaskDefinitionResponse, error) {
```

Comment:
If we can avoid adding this that'll be good, as any plugin that uses it is making a mistake. See comment above...